include/v8-object.h

```diff
@@ -232,6 +232,46 @@ enum class KeyConversionMode { kConvertToString, kKeepNumbers, kNoNumbers };
  */
 enum class IntegrityLevel { kFrozen, kSealed };
 
+using KeyIterationIndex = int;
+
+/**
+ * [RecordReplay]
+ * Used to paginate keys for performance reasons (RUN-1315).
+ * This is currently only implemented as part of the debugger's `Runtime.getProperties`
+ * command. It is also not a guarantee on pagination, but purely a performance improvement.
+ * We apply these params in many scenarios where large key sets would slow things down a lot.
+ * NOTE: In the future, this can be updated to encapsulate all key query parameters
+ * in a single object (`KeyCollectionMode`, `IndexFilter` etc.).
+ */
+class KeyIterationParams {
+ public:
+  KeyIterationIndex page_size_;
+  KeyIterationIndex page_index_;
+  
+  KeyIterationParams(KeyIterationIndex pageSize, KeyIterationIndex pageIndex)
+      : page_size_(pageSize) , page_index_(pageIndex) {}
+
+  KeyIterationIndex pageSize(KeyIterationIndex numberOfElements) const {
+    return ((bool)*this && page_size_ < numberOfElements) ? page_size_
+                                                         : numberOfElements;
+  }
+
+  KeyIterationIndex keyFirstIndex() const { return page_index_ * page_size_; }
+
+  KeyIterationIndex keyEndIndex(KeyIterationIndex numberOfElements) const {
+    return ((bool)*this && (keyFirstIndex() + page_size_) < numberOfElements)
+               ? keyFirstIndex() + page_size_
+               : numberOfElements;
+  }
+
+  operator bool() const { return page_size_ > 0; }
```

Comment:
Implicit methods like this aren't great, can this be an explicit member function?

Reply:
I'm not sure I would say they "arent' great". That's rather subjective 👀 
Can you provide some arguments?

Reply:
In general I don't like C++ that contains implicit behavior, though there are exceptions.  It makes things hard to grep for and understand -- I shouldn't have to look at the disassembly for a function to figure out what it is going to do.

Reply:
In general I don't like C++ that contains implicit behavior, though there are exceptions.  It makes things hard to grep for and understand -- I shouldn't have to look at the disassembly for a function to figure out what it is going to do.

Reply:
Ok, I'll apply that reasoning next time.
I find it to be quite a powerful tool, but I can see how the lack of expliciteness can be frustrating.